src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs

```diff
@@ -232,6 +233,55 @@ public TagHelperContent EndTagHelperWritingScope()
         }
 
         /// <summary>
+        /// Starts a new scope for writing <see cref="ITagHelper"/> attribute values.
+        /// </summary>
+        /// <remarks>
+        /// All writes to the <see cref="Output"/> or <see cref="ViewContext.Writer"/> after calling this method will
+        /// be buffered until <see cref="EndWriteTagHelperAttribute"/> is called.
+        /// The content will be buffered using a shared <see cref="StringWriter"/> within this <see cref="RazorPage"/> 
+        /// This method assumes that there will be no nesting of <see cref="BeginWriteTagHelperAttribute"/> 
+        /// and <see cref="EndWriteTagHelperAttribute"/> method calls.
+        /// </remarks>
+        public void BeginWriteTagHelperAttribute()
+        {
+            _pageWriter = ViewContext.Writer;
+
+            if (_valueBuffer == null)
+            {
+                _valueBuffer = new StringWriter();
+            }
+
+            // We need to replace the ViewContext's Writer to ensure that all content (including content written
+            // from HTML helpers) is redirected.
+            ViewContext.Writer = _valueBuffer;
+
+        }
+
+        /// <summary>
+        /// Ends the current writing scope that was started by calling <see cref="BeginWriteTagHelperAttribute"/>.
+        /// </summary>
+        /// <returns>The content buffered by the shared <see cref="StringWriter"/> of this <see cref="RazorPage"/>.</returns>
+        /// <remarks>
+        /// This method assumes that there will be no nesting of <see cref="BeginWriteTagHelperAttribute"/> 
+        /// and <see cref="EndWriteTagHelperAttribute"/> method calls.
+        /// </remarks>
+        public string EndWriteTagHelperAttribute()
+        {
+            if (_pageWriter == null)
+            {
+                throw new InvalidOperationException(Resources.RazorPage_ThereIsNoActiveWritingScopeToEnd);
+            }
+
+            var content = _valueBuffer.ToString();
```

Comment:
Is `ToString`ing here effectively the same as calling `_valueBuffer.GetStringBuilder().ToString()`? Just ensuring this doesn't build up extra assets required to `ToString` the writer.